### ISSUE TITLE
'name' attribute is now required on Font Face definitions (breaking many existing themes)

### ISSUE THREAD

**Issue by pbking** [WordPress/gutenberg#54605]
### Description

Previously the `name` attribute was not required in order for fonts to render.  

Now, when a theme.json is missing the 'name' attribute the font will not render as expected (it is excluded from the resources in `<style id='wp-fonts-local'` element).

A theme.json file used to be able to be shaped like this:

```
				{
					"fontFace": [
						{
							"fontFamily": "Cardo",
							"fontStyle": "normal",
							"fontWeight": "400",
							"src": [
								"file:./assets/fonts/cardo_normal_400.woff2"
							]
						}
					],
					"fontFamily": "Cardo",
					"slug": "heading"
				}
```

wheras now it must be shaped like this

```
				{
					"fontFace": [
						{
							"fontFamily": "Cardo",
							"fontStyle": "normal",
							"fontWeight": "400",
							"src": [
								"file:./assets/fonts/cardo_normal_400.woff2"
							]
						}
					],
					"fontFamily": "Cardo",
					"name": "Cardo",
					"slug": "heading"
				}
```

While many themes DO have the name attribute as expected many do not.  Those themes will no longer display correctly once this change is released.

### Step-by-step reproduction instructions

Install and activate a theme that includes fonts but does not include a 'name' attribute.  Currently TwentyTwentyFour is in that state (though that is likely to change soon).  Another option is the [CTLG](https://wordpress.org/themes/ctlg/) theme.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jeffikus** [comment]
Twenty Twenty Four PR https://github.com/WordPress/twentytwentyfour/pull/422

**Comment by bph** [comment]
I close this as it seems to be solved by #54615
